ocaml/testsuite/tests/typing-layouts/mixed_constructor_arguments_alpha.ml

```diff
@@ -0,0 +1,391 @@
+    int * float#
+[%%expect{|
+type ptr = string
+Lines 3-36, characters 2-16:
+ 3 | ..A of
+ 4 |     ptr * ptr * ptr * ptr * ptr * ptr * ptr * ptr *
+ 5 |     ptr * ptr * ptr * ptr * ptr * ptr * ptr * ptr *
+ 6 |     ptr * ptr * ptr * ptr * ptr * ptr * ptr * ptr *
+ 7 |     ptr * ptr * ptr * ptr * ptr * ptr * ptr * ptr *
+...
+33 |     ptr * ptr * ptr * ptr * ptr * ptr * ptr * ptr *
+34 |     ptr * ptr * ptr * ptr * ptr * ptr * ptr * ptr *
+35 |     ptr * ptr * ptr * ptr * ptr * ptr * ptr *
+36 |     int * float#
+Error: Mixed constructors may contain at most 254 value fields prior to the flat suffix, but this one contains 255.
+|}];;
+
+type t_ext +=
+  A of
+    ptr * ptr * ptr * ptr * ptr * ptr * ptr * ptr *
+    ptr * ptr * ptr * ptr * ptr * ptr * ptr * ptr *
+    ptr * ptr * ptr * ptr * ptr * ptr * ptr * ptr *
+    ptr * ptr * ptr * ptr * ptr * ptr * ptr * ptr *
+    ptr * ptr * ptr * ptr * ptr * ptr * ptr * ptr *
+    ptr * ptr * ptr * ptr * ptr * ptr * ptr * ptr *
+    ptr * ptr * ptr * ptr * ptr * ptr * ptr * ptr *
+    ptr * ptr * ptr * ptr * ptr * ptr * ptr * ptr *
+    ptr * ptr * ptr * ptr * ptr * ptr * ptr * ptr *
+    ptr * ptr * ptr * ptr * ptr * ptr * ptr * ptr *
+    ptr * ptr * ptr * ptr * ptr * ptr * ptr * ptr *
+    ptr * ptr * ptr * ptr * ptr * ptr * ptr * ptr *
+    ptr * ptr * ptr * ptr * ptr * ptr * ptr * ptr *
+    ptr * ptr * ptr * ptr * ptr * ptr * ptr * ptr *
+    ptr * ptr * ptr * ptr * ptr * ptr * ptr * ptr *
+    ptr * ptr * ptr * ptr * ptr * ptr * ptr * ptr *
+    ptr * ptr * ptr * ptr * ptr * ptr * ptr * ptr *
+    ptr * ptr * ptr * ptr * ptr * ptr * ptr * ptr *
+    ptr * ptr * ptr * ptr * ptr * ptr * ptr * ptr *
+    ptr * ptr * ptr * ptr * ptr * ptr * ptr * ptr *
+    ptr * ptr * ptr * ptr * ptr * ptr * ptr * ptr *
+    ptr * ptr * ptr * ptr * ptr * ptr * ptr * ptr *
+    ptr * ptr * ptr * ptr * ptr * ptr * ptr * ptr *
+    ptr * ptr * ptr * ptr * ptr * ptr * ptr * ptr *
+    ptr * ptr * ptr * ptr * ptr * ptr * ptr * ptr *
+    ptr * ptr * ptr * ptr * ptr * ptr * ptr * ptr *
+    ptr * ptr * ptr * ptr * ptr * ptr * ptr * ptr *
+    ptr * ptr * ptr * ptr * ptr * ptr * ptr * ptr *
+    ptr * ptr * ptr * ptr * ptr * ptr * ptr * ptr *
+    ptr * ptr * ptr * ptr * ptr * ptr * ptr * ptr *
+    ptr * ptr * ptr * ptr * ptr * ptr * ptr * ptr *
+    ptr * ptr * ptr * ptr * ptr * ptr * ptr *
+    int * float#
+[%%expect{|
+Lines 2-35, characters 2-16:
+ 2 | ..A of
+ 3 |     ptr * ptr * ptr * ptr * ptr * ptr * ptr * ptr *
+ 4 |     ptr * ptr * ptr * ptr * ptr * ptr * ptr * ptr *
+ 5 |     ptr * ptr * ptr * ptr * ptr * ptr * ptr * ptr *
+ 6 |     ptr * ptr * ptr * ptr * ptr * ptr * ptr * ptr *
+...
+32 |     ptr * ptr * ptr * ptr * ptr * ptr * ptr * ptr *
+33 |     ptr * ptr * ptr * ptr * ptr * ptr * ptr * ptr *
+34 |     ptr * ptr * ptr * ptr * ptr * ptr * ptr *
+35 |     int * float#
+Error: Mixed constructors may contain at most 254 value fields prior to the flat suffix, but this one contains 255.
+|}];;
```

Comment:
I'd like to have tests for gadt syntax.  Both along the lines of the tests  you already have, but then also some where the layout is different in the different branches.  Testing quickly, this is allowed:
```ocaml
type ('a : float64) tf
type ('a : value) tv

type ('a : any) t = A : 'a tf -> 'a t
                  | B : 'b tv -> 'a t
```
But this variation on t is rejected:
```ocaml
type ('a : any) t = A : float# * 'a tf -> 'a t
                  | B : 'b tv * float# -> 'a t
```
why?

Reply:
I think the second `t` should certainly be rejected. `'a tf` is itself a value (though its argument is float64).

Reply:
I've added some tests, including a version of the test you proposed with `type ('a : float64) tf : float64` instead.

Reply:
Oops!  Thanks for adding some gadt tests, and sorry for the noise in this example.

(I of course meant to write
```ocaml
type ('a : float64) tf = 'a
type ('a : value) tv = 'a
```
which works as expected)